clients/src/main/java/org/apache/kafka/common/serialization/Serdes.java

```diff
@@ -16,93 +16,103 @@
 import org.apache.kafka.common.utils.Bytes;
 
 import java.nio.ByteBuffer;
+import java.util.Map;
 
 /**
  * Factory for creating serializers / deserializers.
  */
 public class Serdes {
 
-    static public final class LongSerde implements Serde<Long> {
-        @Override
-        public Serializer<Long> serializer() {
-            return new LongSerializer();
-        }
+    static private class WrapperSerde<T> implements Serde<T> {
+        protected Serializer<T> serializer;
+        protected Deserializer<T> deserializer;
 
-        @Override
-        public Deserializer<Long> deserializer() {
-            return new LongDeserializer();
+        public WrapperSerde(Serializer<T> serializer, Deserializer<T> deserializer) {
+            this.serializer = serializer;
+            this.deserializer = deserializer;
         }
-    }
 
-    static public final class IntegerSerde implements Serde<Integer> {
-        @Override
-        public Serializer<Integer> serializer() {
-            return new IntegerSerializer();
+        protected WrapperSerde() {
+            serializer = null;
+            deserializer = null;
         }
 
         @Override
-        public Deserializer<Integer> deserializer() {
-            return new IntegerDeserializer();
+        public void configure(Map<String, ?> configs, boolean isKey) {
+            serializer.configure(configs, isKey);
+            deserializer.configure(configs, isKey);
         }
-    }
 
-    static public final class DoubleSerde implements Serde<Double> {
         @Override
-        public Serializer<Double> serializer() {
-            return new DoubleSerializer();
+        public void close() {
+            serializer.close();
```

Comment:
We do not typically use `assert` in Kafka src code, and in test code we use `JUnit assertXX`. I agree with @mjsax that we do not need to check `null` since it is not exposed to user.